examples/src/components/AttributeFilterExample.jsx

```diff
@@ -1,26 +1,114 @@
 // (C) 2007-2019 GoodData Corporation
 import React, { Component } from "react";
-import { AttributeFilter } from "@gooddata/react-components";
+import { AfmComponents, AttributeFilter, ErrorComponent } from "@gooddata/react-components";
 
 import "@gooddata/react-components/styles/css/main.css";
 
-import { employeeNameIdentifier, projectId } from "../utils/fixtures";
+import {
+    totalSalesIdentifier,
+    locationResortIdentifier,
+    locationResortUri,
+    projectId,
+} from "../utils/fixtures";
+
+const { LineChart } = AfmComponents;
 
 export class AttributeFilterExample extends Component {
-    onApply(params) {
+    constructor(props) {
+        super(props);
+
+        this.onApply = this.onApply.bind(this);
+        this.state = {
+            filters: [],
+            error: null,
+        };
+    }
+
+    onLoadingChanged(...params) {
         // eslint-disable-next-line no-console
-        console.log("AttributeFilterExample onApply", ...params);
+        console.info("AttributeFilterExample onLoadingChanged", ...params);
+    }
+
+    onApply(filter) {
+        // eslint-disable-next-line no-console
+        console.log("AttributeFilterExample onApply", filter);
+        const isPositive = !!filter.in;
+        const elementsProp = isPositive ? "in" : "notIn";
+        this.setState({ error: null });
+        if (filter[elementsProp].length !== 0) {
+            const filters = [
+                {
+                    [isPositive ? "positiveAttributeFilter" : "negativeAttributeFilter"]: {
```

Comment:
I would extract this in the same way as `elementsProp`. Ideally, put the code to separate methods that build an either positive or negative filter and use the first or the second private method based on the `isPositive` value. This would have to be refactored in case we add some new properties to one filter type but not the other. It would be also more readable.